packages/xl-docx-exporter/src/docx/util/Table.tsx

```diff
@@ -12,33 +19,79 @@ export const Table = (
   t: Exporter<any, any, any, any, ParagraphChild, any, any>
 ) => {
   const DEFAULT_COLUMN_WIDTH = 120;
+
+  // If headerRows is 1, then the first row is a header row
+  const headerRows = new Array(data.headerRows ?? 0).fill(true);
+
   return new DocxTable({
     layout: "autofit",
     columnWidths: data.columnWidths.map(
       (w) =>
         (w ?? DEFAULT_COLUMN_WIDTH) * /* to points */ 0.75 * /* to twips */ 20
     ),
-    rows: data.rows.map(
-      (row) =>
-        new TableRow({
-          children: row.cells.map((cell, i) => {
-            const width = data.columnWidths?.[i];
-            return new TableCell({
-              width: width
-                ? {
-                    size: `${width * 0.75}pt`,
-                    type: "dxa",
-                  }
-                : undefined,
-              children: [
-                new Paragraph({
-                  // TODO: fix this
-                  children: t.transformInlineContent(cell as any),
-                }),
-              ],
-            });
-          }),
-        })
-    ),
+    rows: data.rows.map((row, rowIndex) => {
+      const isHeaderRow = headerRows[rowIndex];
+      return new TableRow({
+        tableHeader: isHeaderRow,
+        children: row.cells.map((c, colIndex) => {
+          const width = data.columnWidths?.[colIndex];
+          const cell = mapTableCell(c);
+
+          return new TableCell({
+            width: width
+              ? {
+                  size: `${width * 0.75}pt`,
+                  type: "dxa",
+                }
+              : undefined,
+            columnSpan: cell.props.colspan,
+            rowSpan: cell.props.rowspan,
+            shading:
+              cell.props.backgroundColor === "default" ||
+              !cell.props.backgroundColor
+                ? undefined
+                : {
+                    type: ShadingType.SOLID,
+                    color:
+                      t.options.colors[
+                        cell.props
+                          .backgroundColor as keyof typeof t.options.colors
+                      ].background.slice(1),
+                  },
+            children: [
+              new Paragraph({
+                children: t.transformInlineContent(cell.content),
+
+                alignment:
+                  !cell.props.textAlignment ||
+                  cell.props.textAlignment === "left"
+                    ? undefined
+                    : cell.props.textAlignment === "center"
+                    ? "center"
+                    : cell.props.textAlignment === "right"
+                    ? "right"
+                    : cell.props.textAlignment === "justify"
+                    ? "distribute"
+                    : (() => {
+                        throw new UnreachableCaseError(
+                          cell.props.textAlignment
+                        );
+                      })(),
+                run:
```

Comment:
- you could add "bold" here as well for headers

My guess why it's not working might be because the children of the paragraph overwrite all the styles. You could break / log here to confirm: https://github.com/TypeCellOS/BlockNote/blob/11c492146fbb7819586e994d046a7f70a4330179/packages/xl-docx-exporter/src/docx/docxExporter.ts#L98

If that's not the case, then maybe the styles don't inherit the nested runs and we'd need to find another way.

lmk if you need help with digging in further